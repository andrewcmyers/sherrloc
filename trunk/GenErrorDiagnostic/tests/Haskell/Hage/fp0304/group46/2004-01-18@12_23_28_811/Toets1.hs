module Toets1 where

verwijderRec :: (Int -> Int -> Bool) -> Int -> [Int] -> [Int]
verwijderRec test n []     = []
verwijderRec test n (x:xs) | x==n = verwijderRec test n xs
                           | x/=n = x:verwijderRec test n xs

verwijderStFun :: (Int -> Int -> Bool) -> Int -> [Int] -> [Int]
verwijderStFun test n [] = []
verwijderStFun test n (x:xs) = filter (n/=) (x:xs)

groepeer :: Int -> [Int] -> [[Int]]
groepeer n [] = []
groepeer n (x:xs) = (x:take (n-1) xs)

-- signature should be: Int -> [Int] -> [Int]
-- 12,13-35   12,29-35   12,1-8
