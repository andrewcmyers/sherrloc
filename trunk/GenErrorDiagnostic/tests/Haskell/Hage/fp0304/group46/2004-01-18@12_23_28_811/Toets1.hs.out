Compiling Toets1.hs
(14,22)-(14,37): Type error in right-hand side
 expression       : x : take (n - 1) xs
   type           : [Int  ]
   does not match : [[Int]]

Compilation failed with 1 error
